core-image.html

```diff
@@ -179,12 +212,94 @@
        * @type number
        * @default null
        */
-      height: null
+      height: Number
+    },
+
+    configure: function() {
+      return {
+        src: '',
+        preventLoad: false,
+        preload: false,
+        loading: false,
+        loaded: false,
+        placeholder: null,
+        fade: false,
+        sizing: null,
+        position: 'center',
+        width: null,
+        height: null,
+      };
+    },
+
+    bind: {
+      width: 'widthChanged',
+      height: 'heightChanged',
+      src: 'srcChanged',
+      placeholderBackgroundUrl: 'placeholderBackgroundUrlChanged'
+    },
+
+    computed: {
+      imageClassName: 'computeImageClassName(sizing)',
+      placeholderClassName: 'computePlaceholderClassName(fade,loaded,preload)',
+      placeholderBackgroundUrl: 'computePlaceholderBackgroundUrl(preload,placeholder)',
+      requiresPreload: 'computeRequiresPreload(preload,loaded)',
+      canLoad: 'computeCanLoad(preventLoad,src)',
+
+      // TODO(cdata): Change this to a bind when we can declare multiple
+      // bind dependencies in 0.8
+      _observe1: 'transformChanged(sizing,position)',
+      _observe2: 'loadBehaviorChanged(canLoad,preload,loaded)',
+      _observe3: 'loadStateChanged(src,preload,loaded)',
+    },
+
+    ready: function() {
+      if (!this.hasAttribute('role')) {
+        this.setAttribute('role', 'img');
+      }
+    },
+
+    computeImageClassName: function() {
+      if (this.sizing) {
+        return 'hidden';
+      }
+
+      return '';
+    },
+
+    computePlaceholderClassName: function() {
+      var className
+        
+      if (!this.preload) {
+        return 'hidden';
+      }
 
+      className = 'fit';
+
+      if (this.loaded) {
+        if (this.fade) {
+          className += ' faded-out';
+        } else {
+          className += ' hidden';
+        }
+      }
+
+      return className;
+    },
+
+    computePlaceholderBackgroundUrl: function() {
+      if (this.preload && this.placeholder) {
+        return 'url(' + this.placeholder + ')';
+      }
+
+      return null;
+    },
+
+    computeRequiresPreload: function() {
+      return this.preload && !this.loaded;
     },
 
-    observe: {
-      'preload color sizing position src fade': 'update'
+    computeCanLoad: function() {
+      return !(this.preventLoad || !this.src);
```

Comment:
For legibility does it make sense to distribute the negation? Less to do in my head the next time around:

    return !this.preventLoad && this.src;